fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1153,6 +1135,7 @@ def _init_param_attributes(self, p: Parameter) -> None:
             return
 
         # A single shard of the parameters in full precision.
+        # PJ this will cause memory leakage with ssd
```

Comment:
Is there a followup TODO here or just a caution for readers? We use properties of p to set other attributes but don't deal with the data itself. Wondering what kind of guard we can add here to prevent errors. Maybe we should add a TODO to followup :)

Reply:
My understanding is that 1139:
```
p._fp32_shard = p.data
```

Will invoke SsdFlatParam's torch_dispatch function, returning a tensor. If we save a ptr to this tensor (p._fp32_shard), then this tensor will never be freed by the GC. Even if we call p.to_file(), the refcount to the tensor will still be 1 because of p._fp32_shard will still be pointing to it.